extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/vertx/JsonArrayWriter.java

```diff
@@ -35,4 +38,14 @@ public void writeTo(JsonArray jsonArray, Class<?> type, Type genericType, Annota
         entityStream.flush();
         entityStream.close();
     }
+
+    @Override
+    public CompletionStage<Void> asyncWriteTo(JsonArray jsonArray, Class<?> type, Type genericType, Annotation[] annotations,
+            MediaType mediaType,
+            MultivaluedMap<String, Object> httpHeaders,
+            AsyncOutputStream entityStream) {
+        CompletionStage<Void> a = entityStream.asyncWrite(jsonArray.toBuffer().getBytes());
+        CompletionStage<Void> b = entityStream.asyncFlush();
+        return a.thenCompose(v -> b);
```

Comment:
Can't `asyncFlush` happen before `asyncWrite` in this snippet? Wondering if it's better to read:

```suggestion
        return entityStream.asyncWrite(jsonArray.toBuffer().getBytes())
                       .thenCompose(v -> entityStream.asyncFlush());
```

Reply:
I just removed the flush, it looks like none of the other writers have it, and in general flushes are a bad idea as they can force chunked encoding when it is not nessesary.